src/main/java/com/zendesk/maxwell/schema/columndef/ColumnDefWithLength.java

```diff
@@ -65,13 +52,15 @@ protected static String appendFractionalSeconds(String value, int nanos, Long co
 		int divideBy = (int) Math.pow(10, 6 + 3 - columnLength);
 		int fractional = nanos / divideBy;
 
-		String strFractional = String.format(buildStrFormatForColLength(columnLength), fractional);
-
 		StringBuilder result = threadLocalBuilder.get();
 		result.append(value);
 		result.append(".");
-		result.append(strFractional);
 
+		String asStr = String.valueOf(fractional);
+		for ( int i = 0 ; i < columnLength - asStr.length(); i++ )
+			result.append("0");
+
+		result.append(asStr);
```

Comment:
Looks good! Though a unit test will help to iron out some boundary issues.

Reply:
bunch of unit tests in `ColumnDefTest` already...

Reply:
👍